amundsen_application/static/js/components/common/SearchBar/index.tsx

```diff
@@ -92,12 +92,14 @@ export class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
   handleValueChange = (event: React.SyntheticEvent<HTMLInputElement>) : void => {
     const searchTerm = (event.target as HTMLInputElement).value.toLowerCase();
 
-    if (searchTerm.length > 0) {
-      this.props.onInputChange(searchTerm);
-      this.setState({ searchTerm, showTypeAhead: true });
-    }
-    else {
-      this.clearSearchTerm();
+    if (searchTerm.indexOf(':') < 0) {
+      if (searchTerm.length > 0) {
+        this.props.onInputChange(searchTerm);
+        this.setState({ searchTerm, showTypeAhead: true });
+      }
+      else {
+        this.clearSearchTerm();
+      }
```

Comment:
Might be simpler to use a keypress handler and check `event.code` to see if a `:` was typed and then `event.preventDefault` to stop the `:` character from being added. This approach adds the character, then removes it after the value is updated.

Reply:
Whats your thought on letting them type the ":" then showing an error message + blocking submit, vs the keypress approach?

Reply:
On second thought I like showing the error message better since it lets the user know what's going on rather than just blocking them and making them think it's a bug.